cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1514,14 +1514,16 @@ void reader::impl::preprocess_subpass_pages(read_mode mode, size_t chunk_read_li
       max_col_row--;
     }
 
-    max_row = min(max_row, max_col_row);
+    max_row = std::min<size_t>(max_row, max_col_row);
 
     first_page_index += subpass.column_page_count[idx];
   }
   subpass.skip_rows   = pass.skip_rows + pass.processed_rows;
   auto const pass_end = pass.skip_rows + pass.num_rows;
-  max_row             = min(max_row, pass_end);
-  subpass.num_rows    = max_row - subpass.skip_rows;
+  max_row             = std::min<size_t>(max_row, pass_end);
+  // Make sure we don't skip past the max rows.
+  CUDF_EXPECTS(max_row > subpass.skip_rows, "Unexpected subpass row count");
```

Comment:
Fail here predictably instead of throwing unhandled exceptions or going into a forever loop.